lectures/5-Tutorials/SymmetricTensors.md

```diff
@@ -0,0 +1,1240 @@
+construct them as `TensorMap`s using an *auxiliary vector space*, based on the following
+intuition. The creation operator $a^+$ violates particle number conservation by mapping the
+occupation number $n$ to $n + 1$. From the point of view of representation theory, this
+process can be thought af the *fusion* of an `U1Irrep(n)` with an `U1Irrep(1)`, naturally
+giving the fusion product `U1Irrep(n + 1)`. This means we can represent $a^+$ as a
+`TensorMap(..., V ← V ⊗ A)`, where the auxiliary vector space `A` is contains the $+1$ irrep
+with degeneracy 1, `A = U1Space(1 => 1)`. Similarly, the decrease in occupation number when
+acting with $a^-$ can be thought of as the *splitting* of an `U1Irrep(n)` into an
+`U1Irrep(n - 1)` and an `U1Irrep(1)`, leading to a representation in terms of a
+`TensorMap(..., A ⊗ V ← V)`. Based on these observations, we can represent the matrix
+elements {eq}`eq:bosonopmatel` as blocks labeled by the $\mathrm{U}(1)$ fusion trees
+
+```{figure} ../_static/SymmetricTensors/bosonops.svg
+:scale: 12%
+:name: bosonops
+```
+
+We can then combine these operators in to get the appropriate Hamiltonian terms,
+
+```{figure} ../_static/SymmetricTensors/bosonham.svg
+:scale: 12%
+:name: bosonham
+```
+
+```{note}
+Although we have made a suggestive distinction between the 'left' and 'right' versions of
+the operators $a_L^\pm$ and $a_R^\pm$, one can actually be obtained from the other by
+permuting the physical and auxiliary indices of the corresponding `TensorMap`s. This
+permutation has no effect on the actual array blocks of the tensors due to the
+[bosonic braiding style](https://jutho.github.io/TensorKit.jl/latest/lib/sectors/#TensorKit.BraidingStyle)
+of $\mathrm{U}(1)$ irreps, so the left and right operators can in essence by seen as the
+'same' tensors. This is no longer the case when considering fermionic systems, where
+permuting indices can in fact change the array blocks as we will see next. As consequence,
+it is much less clear how to construct two-site symmetric operators in terms of local
+symmetric objects.
+```
+
+The explicit construction then looks something like
+
+```{code-cell} julia
+A = U1Space(1 => 1)
+```
+
+```{code-cell} julia
+:tags: [hide-output]
+a⁺ = TensorMap(zeros, ComplexF64, V ← V ⊗ A)
+for (s, f) in fusiontrees(a⁺)
+    a⁺[s, f] .= sqrt(f.uncoupled[1].charge+1)
+end
+a⁺
+```
+
+```{code-cell} julia
+:tags: [hide-output]
+a⁻ = TensorMap(zeros, ComplexF64, A ⊗ V ← V)
+for (s, f) in fusiontrees(a⁻)
+    a⁻[s, f] .= sqrt(f.uncoupled[1].charge)
+end
+a⁻
+```
+
+It is then simple to check that this is indeed what we expect.
+```{code-cell} julia
+@tensor a⁺a⁻_bis[-1 -2; -3 -4] := a⁺[-1; -3 1] * a⁻[1 -2; -4]
+@tensor a⁻a⁺_bis[-1 -2; -3 -4] := a⁻[1 -1; -3] * a⁺[-2; -4 1]
+@tensor N_bis[-1 ; -2] := a⁺[-1; 1 2] * a⁻[2 1; -2]
+
+@test norm(a⁺a⁻_bis - a⁺a⁻) ≈ 0 atol=1e-14
+@test norm(a⁻a⁺_bis - a⁻a⁺) ≈ 0 atol=1e-14
+@test norm(N_bis - N) ≈ 0 atol=1e-14
```

Comment:
```suggestion
@test a⁺a⁻_bis ≈ a⁺a⁻ atol=1e-14
@test a⁻a⁺_bis ≈ a⁻a⁺ atol=1e-14
@test N_bis ≈ N atol=1e-14
```
I think within the @test macro you can just check for equality like this, and avoid the norms alltogether, alternatively just write out `isapprox` should also work, and might be a little more explicit than comparing norms (should result in same code though)